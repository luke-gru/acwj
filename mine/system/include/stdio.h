#ifndef _STDIO_H_
#define _STDIO_H_

#include <stddef.h>
#include <stdarg.h>

#define EOF (-1)

// This FILE definition will do for now
typedef char *FILE;
extern FILE *stderr;
extern FILE *stdout;

FILE *fopen(char *pathname, char *mode);
size_t fread(void *ptr, size_t size, size_t nmemb, FILE *stream);
size_t fwrite(void *ptr, size_t size, size_t nmemb, FILE *stream);
int fclose(FILE *stream);
int printf(char *format);
int fprintf(FILE *stream, char *format);
size_t getline(char **lineptr, size_t *n, FILE *stream);

FILE *popen(const char *command, const char *type);
int pclose(FILE *stream);

int vfprintf(FILE *stream, const char *format, va_list ap);

#endif	// _STDIO_H_
