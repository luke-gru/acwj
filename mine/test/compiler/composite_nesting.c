#include <stdio.h>
struct Bar {
  int b;
};
struct Foo {
  union {
    int a;
    int b;
  } v;
  struct Bar b;
  int z;
};

struct Foo f;

int main() {
  f.v.a = 10;
  f.b.b = 20;
  f.z = 30;
  printf("f.v.a: %d\n", f.v.a);
  printf("f.b.b: %d\n", f.b.b);
  printf("f.z: %d\n", f.z);
  return (0);
}
