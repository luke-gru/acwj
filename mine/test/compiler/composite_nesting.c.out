f.v.a: 10
f.b.b: 20
f.z: 30
